geps/gep-3792/index.md

```diff
@@ -0,0 +1,190 @@
+* Status: Provisional
+
+(See [status definitions](../overview.md#gep-states).)
+
+## User Story
+
+**[Chihiro] and [Ian] want a way for out-of-cluster Gateways to be able to
+usefully participate in a GAMMA-compliant in-cluster service mesh.**
+
+Historically, API gateways and ingress controllers have often been implemented
+using a Service of type LoadBalancer fronting a Kubernetes pod running a
+proxy. This is simple to reason about, easy to manage for sidecar meshes, and
+will presumably be an important implementation mechanism for the foreseeable
+future. Some cloud providers, though, are moving the proxy outside of the
+cluster, for various reasons which are out of the scope of this GEP. Chihiro
+and Ian want to be able to use these out-of-cluster proxies effectively and
+safely, though they recognize that this may require additional configuration.
+
+[Chihiro]: https://https//gateway-api.sigs.k8s.io/concepts/roles-and-personas/#chihiro
+[Ian]: https://https//gateway-api.sigs.k8s.io/concepts/roles-and-personas/#ian
+
+### Nomenclature and Background
+
+In this GEP, we will use _out-of-cluster Gateway_ (OCG) to refer to a
+conformant implementation of Gateway API's `GATEWAY` profile that's running
+outside of the cluster. This would most commonly be a managed implementation
+from a cloud provider, but of course there are many other possibilities.
+
+We'll also distinguish between _mTLS meshes_, which rely on standard mTLS for
+secure communications including workload authentication, and _non-mTLS
+meshes_, which do anything else. We'll focus on mTLS meshes in this GEP; this
+isn't because of a desire to exclude non-mTLS meshes, but because we'll have
+enough trouble just wrangling the mTLS meshes! Supporting non-mTLS meshes will
+be a separate GEP.
+
+Finally, _authentication_ is the act of verifying the identity of some
+_principal_; what the principal actually is depends on context. For this GEP
+we will primarily be concerned with _workload authentication_ (also known as
+_east/west auth_), in which the principal is a workload. (For completeness'
+sake, in _user authentication_ AKA _N/S auth_, the principal is the human on
+whose behalf a piece of technology is acting. We're only concerned with it
+here in that meshed workloads can't trust what the OCG says about user
+authentication without successful workload auth of the OCG itself.)
+
+## Goals
+
+- Allow Chihiro and Ian to configure an OCG and a mesh such that the OCG can
+  usefully participate in the mesh, including:
+
+    - The OCG must be able to securely communicate with meshed workloads in
+      the cluster, where "securely communicate" includes encryption,
+      authentication, and integrity checking.
+
+    - The OCG must have a proper identity within the mesh, so that the mesh
+      can apply authorization policy to requests from the OCG.
+
+    - Whatever credentials the OCG and the mesh use to authenticate each other
```

Comment:
while meshes may have mtls on by default, it's definetly possible that mtls is turned off for a set of workloads/services.
So shouldn't the OCG automatically pick up on this info and not do MTLS ?